ymer/src/test/java/com/avanza/ymer/DocumentPatchApplyTest.java

```diff
@@ -0,0 +1,160 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import org.bson.Document;
+import org.junit.Test;
+import com.mongodb.BasicDBObject;
+
+public class DocumentPatchApplyTest {
+    @Test
+    public void shouldHandlePatchRemovalsUsingBasicDBObject() {
+
+        // Given
+        DocumentPatch documentPatch$1 = new DocumentPatch() {
+
+            @Override
+            public void apply(BasicDBObject dbObject) {
+                dbObject.put("theKey", "theValue");
+                dbObject.put("theKeyToBeRemoved", "theValueToBeRemoved");
+            }
+
+            @Override
+            public int patchedVersion() {
+                return 0;
+            }
+        };
+
+        DocumentPatch documentPatch$2 = new DocumentPatch() {
+
+            @Override
+            public void apply(BasicDBObject dbObject) {
+                dbObject.put("anotherKey", "anotherValue");
+                dbObject.remove("theKeyToBeRemoved");
+            }
+
+            @Override
+            public int patchedVersion() {
+                return 1;
+            }
+        };
+
+        BasicDBObject initialObject = new BasicDBObject();
+
+        // When
+
+        documentPatch$1.apply(initialObject);
+        documentPatch$2.apply(initialObject);
+
+        // Then
+        assertThat(initialObject.get("theKey"), is("theValue"));
+        assertThat(initialObject.get("anotherKey"), is("anotherValue"));
+        assertThat(initialObject.get("toBeRemoved"), nullValue());
+    }
+
+    @Test
+    public void shouldHandlePatchRemovalsUsingDocument() {
+
+        // Given
+        DocumentPatch documentPatch$1 = new DocumentPatch() {
+
+            @Override
+            public void apply(BasicDBObject dbObject) {
+                dbObject.put("theKey", "theValue");
+                dbObject.put("theKeyToBeRemoved", "theValueToBeRemoved");
+            }
+
+            @Override
+            public int patchedVersion() {
+                return 0;
+            }
+        };
+
+        DocumentPatch documentPatch$2 = new DocumentPatch() {
+
+            @Override
+            public void apply(BasicDBObject dbObject) {
+                dbObject.put("anotherKey", "anotherValue");
+                dbObject.remove("theKeyToBeRemoved");
+            }
+
+            @Override
+            public int patchedVersion() {
+                return 1;
+            }
+        };
+
+        Document initialObject = new Document();
+
+        // When
+
+        documentPatch$1.apply(initialObject);
+        documentPatch$2.apply(initialObject);
+
+        // Then
+        assertThat(initialObject.get("theKey"), is("theValue"));
+        assertThat(initialObject.get("anotherKey"), is("anotherValue"));
+        assertThat(initialObject.get("toBeRemoved"), nullValue());
+    }
+
+    @Test
+    public void shouldHandlePatchRemovalsUsingDocumentOnBasicDBOject() {
```

Comment:
The body of this test seems identical to the test just above.

Reply:
Deleting it